mailbox/store/src/main/java/org/apache/james/mailbox/store/search/ListeningMessageSearchIndex.java

```diff
@@ -76,17 +77,10 @@ public void event(Event event) {
                     EventFactory.AddedImpl added = (EventFactory.AddedImpl) event;
                     final Mailbox mailbox = added.getMailbox();
 
-                    for (MessageUid next : (Iterable<MessageUid>) added.getUids()) {
-                        Iterator<MailboxMessage> messages = factory.getMessageMapper(session).findInMailbox(mailbox, MessageRange.one(next), FetchType.Full, -1);
-                        while (messages.hasNext()) {
-                            MailboxMessage message = messages.next();
-                            try {
-                                add(session, mailbox, message);
-                            } catch (MailboxException e) {
-                                session.getLog().error("Unable to index message " + message.getUid() + " for mailbox " + mailbox, e);
-                            }
-                        }
-
+                    if (added.isSingleMessage()) {

```

Comment:
Hmmm I think here that is the wrong question...

The question is: was this messages already prefetched? And not the condition of generation.

Maybe we should replace the optional by a map of uid to messages, to allow treating several added messages in the way single ones are.